src/Hyperwallet/Util/HyperwalletEncryption.php

```diff
@@ -201,31 +206,24 @@ private function getPublicJwsKey() {
      * @param array $privateKeyData The JWK key data
      * @return RSA
      */
-    private function getPrivateKey($privateKeyData) {
-        $n = $this->keyParamToBigInteger($privateKeyData['n']);
-        $e = $this->keyParamToBigInteger($privateKeyData['e']);
-        $d = $this->keyParamToBigInteger($privateKeyData['d']);
-        $p = $this->keyParamToBigInteger($privateKeyData['p']);
-        $q = $this->keyParamToBigInteger($privateKeyData['q']);
-        $qi = $this->keyParamToBigInteger($privateKeyData['qi']);
-        $dp = $this->keyParamToBigInteger($privateKeyData['dp']);
-        $dq = $this->keyParamToBigInteger($privateKeyData['dq']);
-        $primes = array($p, $q);
-        $exponents = array($dp, $dq);
-        $coefficients = array($qi, $qi);
-        array_unshift($primes, "phoney");
-        unset($primes[0]);
-        array_unshift($exponents, "phoney");
-        unset($exponents[0]);
-        array_unshift($coefficients, "phoney");
-        unset($coefficients[0]);
+    private function getPrivateKey($privateKeyData)
+    {
+        $pemData = RSA::load([
+            'e' => $this->keyParamToBigInteger($privateKeyData['e']),
+            'n' => $this->keyParamToBigInteger($privateKeyData['n']),
+            'd' => $this->keyParamToBigInteger($privateKeyData['d']),
+            'p' => $this->keyParamToBigInteger($privateKeyData['p']),
+            'q' => $this->keyParamToBigInteger($privateKeyData['q']),
+            'dp' => $this->keyParamToBigInteger($privateKeyData['dp']),
+            'dq' => $this->keyParamToBigInteger($privateKeyData['dq']),
+            'qi' => $this->keyParamToBigInteger($privateKeyData['qi']),
+        ]);
+
+        $privateKey = RSA::loadPrivateKey($pemData->toString('PKCS1'));
 
-        $pemData = (new RSA())->_convertPrivateKey($n, $e, $d, $primes, $exponents, $coefficients);
-        $privateKey = new RSA();
-        $privateKey->loadKey($pemData);
         if ($privateKeyData['alg'] == 'RSA-OAEP-256') {
-            $privateKey->setHash('sha256');
-            $privateKey->setMGFHash('sha256');
+//            $privateKey->setHash('sha256');
+//            $privateKey->setMGFHash('sha256');
```

Comment:
phpseclib v3 $privateKey is now a PrivateKey object and not RSA object, it changed completely so i am not sure if i should remove this or not.